WikiStuff.py

```diff
@@ -0,0 +1,123 @@
+        # it'll raise `ValueError` exception
+    except ValueError:
+        return False
+    return True
+
+
+def findWikiUrl(album, artist):
+    found = False
+    query = "{title} {art}".format(art=artist, title=album).replace(" ", "+").replace("&", "and")
+    search = "https://en.wikipedia.org/w/index.php?cirrusUserTesting=glent_m0&sort=relevance&search={query}&title=Special:Search&profile=advanced&fulltext=1&advancedSearch-current=%7B%7D&ns0=1".format(
+        query=query)
+    res = requests.get(search, headers=HEADERS_GET).text
+    soup = BeautifulSoup(res, 'html.parser')
+    tags = [tag for tag in soup.find_all(attrs=WIKI_SEARCH_RESULT_ATTRS)]
+    items_hrefs = ['https://en.wikipedia.org/{}'.format(tag.find_next('a', href=True).get('href')) for tag in tags]
+    print("item_hrefs= {}".format(items_hrefs))
+    for j in items_hrefs:
+        # print("j= "+j)
+        if (j.lower().__contains__(artist.lower())) and j.lower().__contains__(
+                album.lower()) and j.lower().__contains__("album"):
+            return j
+        else:
+            if (j.lower().__contains__(album.lower())) and j.lower().__contains__(artist.lower()):
+                print("2 found the song by {} in: {}".format(artist, j))
+                return j
+            else:
+                if (j.lower().__contains__(album.lower())) and j.lower().__contains__("album"):
+                    print("3 found the song by {} in: {}".format(artist, j))
+                    return j
+                else:
+                    if j.lower().__contains__(album.lower()):
+                        print("4 found the song by {} in: {}".format(artist, j))
+                        return j
+    if not found:
+        done = False
+        newtags = [tag.text for tag in soup.find_all(attrs=WIKI_SEARCH_RESULT_DESCRIPTION_ATTRS)]
+        print(str(newtags))
+        try:
+            if items_hrefs[2].__contains__("album") and not items_hrefs[0].__contains__("album") and not items_hrefs[
+                1].__contains__("album") and str(newtags[2]).lower().__contains__(artist.lower()):
+                print("4.1 found the song by {} in: {}".format(artist, items_hrefs[2]))
+                return items_hrefs[2]
+        except IndexError:
+            print("there wa no other option")
+        try:
+            if not done and items_hrefs[0].lower().__contains__("band") and str(newtags[1]).lower().__contains__(
+                    artist.lower()) or (
+                    items_hrefs[1].lower().__contains__("album") and not items_hrefs[0].lower().__contains__("album")):
+                return items_hrefs[1]
+        except IndexError:
+            print("there wa no other option")
+
+        try:
+            if not done:
+                return items_hrefs[0]
+        except IndexError:
+            return None
+    return None
+
+
+def searchLengthInWiki(url, titles):
+    lengths = []
+    j = 0
+    while j < len(titles):
+        found = False
```

Comment:
why did you make a while loop act like a for loop? read about enumerate and remove that while, j, and tmpTitle thingies